broker.go

```diff
@@ -254,6 +265,47 @@ func (b *Broker) RegisterNode(id NodeID, node Node, opt ...Option) error {
 	return nil
 }
 
+// DeregisterNode will remove a node from the broker, if it is not currently  in use
+// This is useful if RegisterNode was used successfully prior to a failed RegisterPipeline call
+// referencing those nodes
+func (b *Broker) DeregisterNode(ctx context.Context, id NodeID, opt ...Option) error {
```

Comment:
Sorry for the late addition, for pipelines, we seem to have followed the format of naming the funcs `Register___`/
`Remove___`. Do you think we should try to keep this going for the nodes for consistency?

Reply:
Sure!